modules/twinkleblock.js

```diff
@@ -0,0 +1,883 @@
+	},
+	'uw-spoablock' : {
+		autoblock: true,
+		nocreate: true,
+		reason: 'Abusing [[WP:Sock puppetry|multiple accounts]]',
+		summary: 'You have been blocked from editing for abusing [[WP:SOCK|multiple accounts]]'
+	},
+	'uw-ublock' : {
+		expiry: 'infinity',
+		forRegisteredOnly: true,
+		reason: '{{uw-ublock}} <!-- Username violation, soft block -->',
+		reasonParam: true,
+		summary: 'You have been indefinitely blocked from editing because your username is a violation of the [[WP:U|username policy]]'
+	},
+	'uw-ublock-double': {
+		expiry: 'infinity',
+		forRegisteredOnly: true,
+		reason: '{{uw-ublock-double}} <!-- Username closely resembles another user, soft block -->',
+		summary: 'You have been indefinitely blocked from editing because your [[WP:U|username]] is too similar to the username of another Wikipedia user'
+	},
+	'uw-uhblock' : {
+		autoblock: true,
+		expiry: 'infinity',
+		forRegisteredOnly: true,
+		nocreate: true,
+		reason: '{{uw-uhblock}} <!-- Username violation, hard block -->',
+		reasonParam: true,
+		summary: 'You have been indefinitely blocked from editing because your username is a blatant violation of the [[WP:U|username policy]]'
+	},
+	'uw-ublock-famous' : {
+		expiry: 'infinity',
+		forRegisteredOnly: true,
+		reason: '{{uw-ublock-famous}} <!-- Username represents a famous person, soft block -->',
+		summary: 'You have been indefinitely blocked from editing because your [[WP:U|username]] matches the name of a well-known living individual'
+	},
+	'uw-uhblock-double': {
+		autoblock: true,
+		expiry: 'infinity',
+		forRegisteredOnly: true,
+		nocreate: true,
+		reason: '{{uw-ublock-double}} <!-- Username closely resembles another user, hard block -->',
+		summary: 'You have been indefinitely blocked from editing because your [[WP:U|username]] appears to impersonate another established Wikipedia user'
+	},
+	'uw-vaublock' : {
+		autoblock: true,
+		expiry: 'infinity',
+		forRegisteredOnly: true,
+		nocreate: true,
+		pageParam: true,
+		reason: '{{uw-vaublock}} <!-- Username violation, vandalism-only account -->',
+		summary: 'You have been indefinitely blocked from editing because your account is being [[WP:VOA|used only for vandalism]] and your username is a blatant violation of the [[WP:U|username policy]]'
+	},
+	'uw-vblock' : {
+		autoblock: true,
+		nocreate: true,
+		pageParam: true,
+		reason: '[[WP:Vandalism|Vandalism]]',
+		summary: 'You have been blocked from editing for persistent [[WP:VAND|vandalism]]'
+	},
+	'uw-voablock' : {
+		autoblock: true,
+		expiry: 'infinity',
+		forRegisteredOnly: true,
+		nocreate: true,
+		pageParam: true,
+		reason: '[[WP:Vandalism-only account|Vandalism-only account]]',
+		summary: 'You have been indefinitely blocked from editing because your account is being [[WP:VOA|used only for vandalism]]'
+	}
+};
+// supply default edit summaries where necessary
+$.each(Twinkle.block.blockPresetsInfo, function(preset, settings) {
```

Comment:
This code is going to run every time anyone (including non-admins) loads Twinkle, whether or not they are going to use the block module. Why can't this just be done on the fly in the place where this data is actually used?